shared-bindings/gnss/GNSS.c

```diff
@@ -0,0 +1,245 @@
+//|                 print("Longitude: {0:.6f} degrees".format(gps.longitude))"""
+//|
+
+//|     def __init__(self, ):
+//|         """Turn on the GNSS."""
+//|         ...
+//|
+STATIC mp_obj_t gnss_make_new(const mp_obj_type_t *type, size_t n_args, const mp_obj_t *pos_args, mp_map_t *kw_args) {
+    gnss_obj_t *self = m_new_obj(gnss_obj_t);
+    self->base.type = &gnss_type;
+
+    common_hal_gnss_construct(self);
+    return MP_OBJ_FROM_PTR(self);
+}
+
+//|     def deinit(self, ) -> Any:
+//|         """Turn off the GNSS."""
+//|         ...
+//|
+STATIC mp_obj_t gnss_obj_deinit(mp_obj_t self_in) {
+    gnss_obj_t *self = MP_OBJ_TO_PTR(self_in);
+    common_hal_gnss_deinit(self);
+    return mp_const_none;
+}
+MP_DEFINE_CONST_FUN_OBJ_1(gnss_deinit_obj, gnss_obj_deinit);
+
+STATIC void check_for_deinit(gnss_obj_t *self) {
+    if (common_hal_gnss_deinited(self)) {
+        raise_deinited_error();
+    }
+}
+
+//|     def select(self, system: gnss.SatelliteSystem) -> Any:
+//|         """Add specified satellite system to selection for positioning.
+//|
+//|         :param gnss.SatelliteSystem system: satellite system to use"""
+//|         ...
+//|
+STATIC mp_obj_t gnss_obj_select(mp_obj_t self_in, mp_obj_t system_obj) {
+    gnss_obj_t *self = MP_OBJ_TO_PTR(self_in);
+    check_for_deinit(self);
+
+    gnss_satellitesystem_t system = gnss_satellitesystem_obj_to_type(system_obj);
+    common_hal_gnss_select(self, system);
+    return mp_const_none;
+}
+STATIC MP_DEFINE_CONST_FUN_OBJ_2(gnss_select_obj, gnss_obj_select);
+
+//|     def deselect(self, system: gnss.SatelliteSystem) -> Any:
+//|         """Remove specified satellite system from selection for positioning.
+//|
+//|         :param gnss.SatelliteSystem system: satellite system to remove"""
+//|         ...
+//|
+STATIC mp_obj_t gnss_obj_deselect(mp_obj_t self_in, mp_obj_t system_obj) {
+    gnss_obj_t *self = MP_OBJ_TO_PTR(self_in);
+    check_for_deinit(self);
+
+    gnss_satellitesystem_t system = gnss_satellitesystem_obj_to_type(system_obj);
+    common_hal_gnss_deselect(self, system);
+    return mp_const_none;
+}
+STATIC MP_DEFINE_CONST_FUN_OBJ_2(gnss_deselect_obj, gnss_obj_deselect);
+
+//|     def start(self, ) -> Any:
+//|         """Start positioning."""
+//|         ...
+//|
+STATIC mp_obj_t gnss_obj_start(mp_obj_t self_in) {
+    gnss_obj_t *self = MP_OBJ_TO_PTR(self_in);
+    check_for_deinit(self);
+
+    common_hal_gnss_start(self);
+    return mp_const_none;
+}
+MP_DEFINE_CONST_FUN_OBJ_1(gnss_start_obj, gnss_obj_start);
+
+//|     def stop(self, ) -> Any:
+//|         """Stop positioning."""
+//|         ...
+//|
+STATIC mp_obj_t gnss_obj_stop(mp_obj_t self_in) {
+    gnss_obj_t *self = MP_OBJ_TO_PTR(self_in);
+    check_for_deinit(self);
+
+    common_hal_gnss_stop(self);
+    return mp_const_none;
+}
+MP_DEFINE_CONST_FUN_OBJ_1(gnss_stop_obj, gnss_obj_stop);
+
+//|     def update(self, ) -> Any:
```

Comment:
The GPS python driver has this as well but I'm not sure it's useful. Is it critical that latitude, longitude and altitude be from the same exact time?

Reply:
I think if you want to use GNSS data, it's worth to be able to get it with the same timestamp. This will make your calculations using this data more accurate.

Reply:
Ok, is the timestamp available as well? Should update return the latest data if it's being called slower than the GPS update rate?

Reply:
I added a timestamp when positioning data was received.

Update does not return data. This function updates all positioning data in the GNSS object to the latest.

I can add that the update function returns bool. When it returns True means that the data has been updated, if False means that the data has not been updated. What do you think about it?

Reply:
I wouldn't object to it but think having `timestamp` is enough too. Go ahead and follow up with another PR if you'd like to.